### DISCUSSION TITLE
Marker durations?

### DISCUSSION THREAD

**Discussion by gkertai** [videovillage/community#107] [Ideas]
I was going to use Screen to prepare a list of notes to my colorist using the markers feature. I did set the marker duration for some, but the lack of visual feedback makes it really hard to use, so eventually I gave up.

For one, the duration of the marker is not indicated on the timeline, which renders that text-field kinda useless. Consequently there is no way to drag the "right edge" of the marker to make it longer/shorter, not even with the timeline zoomed in. Given the similar look-and-feel I was expecting that markers would behave like those of DaVinci Resolve, but they're not nearly as easy to use in Screen.

Another nice touch in Resolve is the heads-up display (overlay) of marker notes currently at the playhead. It provides a convenient way to author and review markers.

So I am wondering… do you have any plans to improve upon the marker UI? I was also considering integration with my workflow tools, but it only makes sense if durations are visually indicated somehow. I can see that you have *Annotations* listed on the [roadmap](https://videovillage.co/screen/) for 2021, which is somewhat related, but not quite the same.

Thanks, and keep up the good work! Screen is an amazing product.

**Comment by gregcotten** [comment]
Yea, the duration of a marker is mainly there because some of the markers file formats have a duration and I didn't want to delete the information on import. Definitely have plans to make this property actually useful! I'll also take a look at the "heads up display" for the markers in Resolve.

**Comment by gkertai** [comment]
Idea for an easy way to add markers with a duration: use I/O to set a range, then ⇧M to "Add Marker for In/Out Range" (I've added a comment about this in videovillage/community#77 too).

With regards to the HUD in DaVince Resolve, it looks like the screenshot below. Not sure if this is applicable to Screen though, but it makes comments really easy to spot/read while playing back a timeline. In Screen you already have the sidebar with the marker text showing, maybe you just need to extend that info and make the sidebar follow playback.

![resolve_marker_hud](https://user-images.githubusercontent.com/1874498/114601571-ee7a8880-9c95-11eb-8da1-ecb033871d5f.png)

**Comment by gkertai** [comment]
I just noticed that DaVinci Resolve has the exact function in its *Mark* menu that I described above… *"Convert In and Out to Duration Marker"* and also the reverse *"Set In and Out from Duration Marker"*.

![resolve_inout_marker](https://user-images.githubusercontent.com/1874498/114621434-fb56a680-9cac-11eb-83ef-e26b96d303b5.png)

I am not at all a Resolve fanboy (hardly ever use it), but it seems like we are not reinventing the wheel. :)

**Comment by gkertai** [comment]
Slightly past the anniversary of this ticket, I happened to be using Screen and its markers for my reviews again. Still no way of converting in/out ranges to a marker, so I thought I'd check in, and give it a bump. 

Hope all is well on your side! Cheers.

**Comment by gregcotten** [comment]
Happy issue anniversary!!! A bit embarrassing, I know. Mainly just didn't know how to make multi-frame duration markers look good in their current context. I'll dig more into it.

**Comment by gkertai** [comment]
No worries, I just wanted to show that I still care. :)

If that's any inspiration, this is how Resolve handles it:

<img width="531" alt="Screen Shot 2022-05-11 at 18 41 38" src="https://user-images.githubusercontent.com/1874498/167903211-adaa10f8-abc4-4948-86e7-8bff41dbabbc.png">

That's a single range, overlapping ranges of the same color, and overlapping ranges of different colors below the timeline ruler. And it uses a more subtle approach on clip markers.

I am more of a Final Cut Pro user, but it does not support marker durations (hence my request here). More precisely, it does support them at the XML level, but they're not differentiated on the UI in any way; not even sure if durations are retained in its internal database.

The only thing similar in FCP is how thumbnail frame positions are displayed for chapter markers, but it's not the most elegant ever, having two different "nodes" at the ends:

<img width="143" alt="Screen Shot 2022-05-11 at 18 47 00" src="https://user-images.githubusercontent.com/1874498/167903997-18e1c017-8252-4a85-9670-f38f92e4581e.png">

Although the dotted connector line is not bad at all.
